test/cli-validator/tests/thresholdValidator.js

```diff
@@ -0,0 +1,89 @@
+// the rule names are all snake case and need to stay that way. don't lint them
+/* eslint-disable camelcase */
+
+const intercept = require('intercept-stdout');
+const expect = require('expect');
+const stripAnsiFrom = require('strip-ansi');
+
+const commandLineValidator = require('../../../src/cli-validator/runValidator');
+
+describe('test the .thresholdrc limits', function() {
+  it('should show error and set exit code to 1 when warning limit exceeded', async function() {
+    const capturedText = [];
+
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/circularRefs.yml'];
+    program.limits = './test/cli-validator/mockFiles/thresholds/.fiveWarnings';
+    program.default_mode = true;
+
+    const unhookIntercept = intercept(function(txt) {
+      capturedText.push(stripAnsiFrom(txt));
+      return '';
+    });
+
+    const exitCode = await commandLineValidator(program);
+
+    unhookIntercept();
+
+    expect(exitCode).toEqual(1);
+
+    expect(capturedText[capturedText.length - 1].slice(0, 18)).toEqual(
+      `Number of warnings`
+    );
+  });
+
+  it('should print errors for unsupported limit options and invalid limit values', async function() {
+    const capturedText = [];
+
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/clean.yml'];
+    program.limits = './test/cli-validator/mockFiles/thresholds/.invalidValues';
+    program.default_mode = true;
+
+    const unhookIntercept = intercept(function(txt) {
+      capturedText.push(stripAnsiFrom(txt));
+      return '';
+    });
+
+    const exitCode = await commandLineValidator(program);
+
+    unhookIntercept();
+
+    // limit values invalid, so default limit, Number.MAX_VALUE, used
+    expect(exitCode).toEqual(0);
+
+    const allOutput = capturedText.join('');
+
+    expect(
+      allOutput.includes('"population" limit not supported.') &&
+        allOutput.includes('Value provided for warnings')
+    ).toEqual(true);
```

Comment:
I would make these separate assertions - best practices say to modularize tests as best you can, and within tests, specific assertions